fem/coefficient.hpp

```diff
@@ -692,13 +692,16 @@ class MatrixCoefficient
 protected:
    int height, width;
    double time;
+   bool symmetric;
```

Comment:
I'm not really convinced by this `symmetric` attribute, I think it would make more sense to implement a `SymmetricMatrixCoefficient`.

Reply:
I first proposed a `SymmetricMatrixCoefficient` but @pazner pointed out that this would require reimplementing all the derived classes for the symmetric case. If you wanted to use a class derived from `MatrixCoefficient`, it would not work if we use a `SymmetricMatrixCoefficient`. With this implementation, one would need to implement the `EvalSymmetric` function in the class and set the symmetry flag, but it would work.